src/child/actions/window.js

```diff
@@ -71,15 +78,22 @@ export function resizeToCompact() {
     };
 }
 
-export function resizeToDefault() {
+export function resizeToPrevious(dimensions, isMaximised) {
     return dispatch => {
         dispatch(resizing());
-        const defaultWindowDimensions = configService.getDefaultWindowDimensions();
         fin.desktop.Window.getCurrent().resizeTo(
-            defaultWindowDimensions[0],
-            defaultWindowDimensions[1],
+            dimensions[0],
+            dimensions[1],
             'top-right',
-            () => dispatch(expand()),
+            () => {
+                dispatch(expand());
+                if (isMaximised) {
```

Comment:
Ah, good point!

Just to check my understanding: if `resizeTo()` isn't called and the user was to do the following:

1. Open app
2. Maximise window
3. Compact view
4. Expand (back to maximised)
5. Restore

Then step 5 would be wrong as it would restore to compact view dimensions, and not expanded dimensions?

If that's the case, then it looks like performing the `resizeTo` would be necessary :+1: